### DISCUSSION TITLE
Can cy.intercept be used for api mocking in vitest or jest

### DISCUSSION THREAD

**Discussion by krishna-as-dev** [cypress-io/cypress#30188] [Questions and Help] [answered]
So in my react app I have written unit and integration test using vitest and e2e test using cypress.
I want to mock apis for my test and cypress provides `cy.intercept` that works for e2e, but I also want to use these api mocks in my vitest and don’t want to install mock service provider (MSW) for the same job.
so is it possible to use cy.intercept and if yes how?

**Comment by portyu9** [comment, marked as answer]
Hi @krishna-as-dev 

No — `cy.intercept()` cannot be used from a normal Vitest or Jest test run.

`cy.intercept()` is part of the Cypress runtime and depends on Cypress's command queue and network interception layer. Vitest/Jest do not run inside that environment.

If you want to avoid duplicating mocks, a good pattern is to share the **mock data/builders**, while keeping the interception mechanism specific to each runner.

For example:

```ts
// test-data/users.ts
export const usersResponse = [
  { id: 1, name: 'Alice' },
]
```

Cypress can use it with:

```ts
cy.intercept('GET', '/api/users', {
  body: usersResponse,
})
```

and Vitest can use the same fixture with its own network-mocking layer.

For Vitest specifically, the current documentation recommends **MSW** for mocking network requests. That is also an option if you want one network-mocking abstraction that can be reused in more environments.

Alternatively, if you move those component/integration tests to **Cypress Component Testing**, then `cy.intercept()` is available because the tests are actually running under Cypress.

No screenshots or attachments were included in this discussion.

Please mark this answer as accepted if it helped, thank you.
